libraries/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -152,6 +162,10 @@ export class ProjectWatcher {
 
             // Need an async tick to allow for more file system events to be handled
             process.nextTick(() => {
+              if (this.isPaused) {
+                this._setStatus(`Project watcher paused. Press 'w' to toggle. Press 'b' to build once.`);
+                return;
+              }
```

Comment:
Do this before line 159 where it evaluates the state change.

What we want to happen is for calling `resume()` to immediately run the `resolveIfChanged` callback, so we'll probably have to persist it onto `this` and document that `waitForChange` is not allowed to be called multiple times concurrently.